### PR TITLE
Added ability to pass relativeTo to routerLink

### PR DESCRIPTION
**Pull request by eyalhakim**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines:  https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
**didn't read this before committing, how do i fix?**

- [ ] Tests for the changes have been added (for bug fixes / features)
**couldn't find tests for routerLink whatsoever**

- [ ] Docs have been added / updated (for bug fixes / features)
**where are the relevant docs located?**

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

routerLink directive always navigate relatively to the current activated route
Issue Number: N/A

## What is the new behavior?
routerLink is now able to navigate relatively to other routes

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by splincode** [comment]
@eyalhakim what is the status of your pull request? is it still relevant?

**Comment by eyalhakim** [comment]
Sure, it's still not implemented as far as I know.

**Comment by markgolding** [comment]
What is the timeline to getting this implemented really need this feature.

**Comment by jasonaden** [comment]
Sorry this has sat so long. We've reviewed our process and are trying to pick up some of these long-standing PRs to get them addressed.

That being said, can you please provide a few things so this can be merged:

* Because this is adding a new public API, we need to have some documentation for the feature. Take a look at the comment block above your change and you can add some documentation there.
* We need to have some tests for this change. You can take a look at the other [`routerLink` tests here](https://github.com/angular/angular/blob/master/packages/router/test/integration.spec.ts#L1458) for examples.
* Rebase to current Angular.

File: packages/router/src/directives/router_link.ts [hunk 1]
```diff
@@ -101,6 +101,7 @@ export class RouterLink {
   @Input() preserveFragment: boolean;
   @Input() skipLocationChange: boolean;
   @Input() replaceUrl: boolean;
+  @Input() relativeTo: ActivatedRoute;
```
Comment (jasonaden) [inline packages/router/src/directives/router_link.ts]: This needs documentation.

Comment (jasonaden) [reply packages/router/src/directives/router_link.ts]: Also a test to confirm it works.

**Review by jasonaden — CHANGES_REQUESTED** [review]

**Pull request closed without merging**